EDU-1026: add Cloud latency info (#2309)
temporalio/documentation: docs-src/cloud/operating-envelope-latency.md

As we continue working on improving latencies, these numbers will decrease over time.

> The latency observed from the Temporal Client might be affected by other system components, such as the Codec Server, egress proxy, and the network itself.

Reviewer: ```suggestion
Other system components, such as the Codec Server, egress proxy, and the network itself, could impact the latency observed from the Temporal Client.
```

passive -> active

> Be aware that the latency observed on the Temporal Client might be impacted by other system components, such as the Codec Server, egress proxy, and the network itself.

Reviewer 2: ```suggestion
The latency observed from the Temporal Client might be impacted by other system components, such as the Codec Server, egress proxy, and the network itself.
```

I like avoiding commands, and to some extent every sentence in the docs is "be aware". If important enough, would add a warning block, but don't think important enough here

> Be aware that the latency observed on the Temporal Client might be impacted by other system components, such as the Codec Server, egress proxy, and the network itself.

Reviewer 3: ```suggestion
Be aware that the latency observed on the Temporal Client might be affected by other system components, such as the Codec Server, egress proxy, and the network itself.
```
Per https://developers.google.com/style/word-list#impact.
